src/coreclr/src/jit/compiler.h

```diff
@@ -8095,6 +8087,7 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
         else
         {
             assert(getSIMDSupportLevel() >= SIMD_SSE2_Supported);
+            compExactlyDependsOn(InstructionSet_AVX2); // Record that AVX2 isn't supported
```

Comment:
This really feels uncomfortable. Might it be worth adding a `compExactlyForbids`? I know we've been over this before, but this looks like an assertion that it is present, not an assertion that it is not.

Reply:
We haven't been over this particular usage before, as it might have been obscured by some of the other changes you were looking at. I share your discomfort here, Perhaps 

```
// Ensure that code will not execute if an instruction set is useable. Call only
// if the instruction set has previously reported as unuseable, but when
// that that status has not yet been recorded to the AOT compiler
void compVerifyInstructionSetUnuseable(CORINFO_InstructionSet isa)
{
    // use compExactlyDependsOn to ensure that the exact state of the useability of isa is recorded
    bool isaUseable = compExactlyDependsOn(isa);
    // Assert that the useability of the isa is false. If true, this function should never be called.
    assert(!isaUseable);
}
```

Reply:
@davidwrighton - that sounds like a great alternative; wordy but certainly a context where a little wordiness is probably justified.